src/modules/push-notifications/push.notifications.service.ts

```diff
@@ -0,0 +1,240 @@
+            this.apiConfigService.getNotificationsApiUrl();
+        const notificationsApiKey =
+            this.apiConfigService.getNotificationsApiKey();
+
+        const payload: NotificationPayload = {
+            addresses,
+            chainId,
+            title: 'Energy Update',
+            body: 'You can now claim your rewards',
+            route: '/portfolio',
+            iconUrl:
+                'https://xexchange.com/assets/imgs/mx-logos/xexchange-logo@2x.webp',
+        };
+
+        const response = await axios.post(notificationsApiUrl, payload, {
+            headers: {
+                'Content-Type': 'application/json',
+                'x-notifications-api-key': notificationsApiKey,
+            },
+        });
+
+        if (response.status !== 201) {
+            return false;
+        }
+
+        return true;
+    }
+
+    private async processNotificationBatch(
+        addresses: string[],
+        maxRetries: number,
+    ): Promise<{ successful: string[]; failed: string[] }> {
+        const batchSize =
+            await this.apiConfigService.getNotificationsBatchSize();
+
+        const successful: string[] = [];
+        let remainingAddresses = [...addresses];
+
+        for (
+            let retryCount = 0;
+            retryCount < maxRetries && remainingAddresses.length > 0;
+            retryCount++
+        ) {
+            if (retryCount > 0) {
+                this.logger.log(
+                    `Retry attempt ${retryCount} for ${remainingAddresses.length} addresses`,
+                );
+                await new Promise((resolve) => setTimeout(resolve, 10000));
+            }
+
+            const addressesToRetry = [...remainingAddresses];
+            remainingAddresses = [];
+
+            for (let i = 0; i < addressesToRetry.length; i += batchSize) {
+                const batch = addressesToRetry.slice(i, i + batchSize);
+                const success = await this.sendBatchNotifications(batch);
+
+                if (success) {
+                    successful.push(...batch);
+                } else {
+                    remainingAddresses.push(...batch);
+                }
+            }
+        }
+
+        return { successful, failed: remainingAddresses };
+    }
+
+    @GetOrSetCache({
+        baseKey: 'notifications',
+        remoteTtl: Constants.oneWeek(),
+        localTtl: Constants.oneDay() * 6,
+    })
+    private async userEnergyAddresses(): Promise<UserEnergyAddress[]> {
+        const isDevnet = process.env.NODE_ENV === 'devnet';
+
+        if (isDevnet) return await this.usersWithEnergyFromContractStorage();
+
+        const currentEpoch = await this.contextGetter.getCurrentEpoch();
+        return await this.getUsersFromIndexer(currentEpoch - 1);
+    }
+
+    @Cron('0 0 19 * * 5') // Every Friday at 19:00
+    @Lock({ name: 'handleNotificationsCron', verbose: true })
+    async handleNotificationsCron() {
+        if (!this.apiConfigService.isNotificationsModuleActive()) {
```

Comment:
No need to have this check as long as this module is only available for notification app